tests/test_native_ops.py

```diff
@@ -188,3 +188,54 @@ def test_apply_amplitude_damp_single_wire(
         simulator_device_1_wire.apply([ops.AmplitudeDamp(*par, wires=[0])])
 
         assert np.allclose(simulator_device_1_wire.state, expected_density_matrix, **tol)
+
+    @pytest.mark.parametrize("input",
+        [
+            [1, 0, 0, 0],
+            [0, 1, 0, 0],
+            [0, 0, 1, 0],
+            [0, 0, 0, 1],
+        ]
+    )
```

Comment:
I struggled to understand this. If the device has two wires, then shouldn't the underlying pure state have shape `(2,2)`?

Reply:
It should have shape `(4,)`, right? For the different bitstrings `00`, `01`, `10` and `11`. Which is what this is (each row corresponding to an input). :thinking:

Reply:
Aha, got it! the pytest parametrize format looks exactly like a 2d array :laughing:

Reply:
Might be good to test on superposition states as well